[mlir][emitc] Clean up EmitC tests

EtoAndruwa opened the pull request:
Changes:
-  Unnecessary `-verify-diagnostics` flags were removed from tests.
- `mlir/test/mlir-translate/emitc_classops.mlir` was moved to `mlir/test/Target/Cpp/class.mlir`.
- The `transfrom.mlir` test was renamed to `form-expressions.mlir` for clarity.
- Test for `emitc.class`, `emitc.field` and `emitc.get_field` was added to `mlir/test/Dialect/EmitC/ops.mlir`.
- `wrap_emitc_func_in_class.mlir` and `wrap_emitc_func_in_class_noAttr.mlir` were combined into `wrap-func-in-class.mlir`.





**---------------------------OLD---------------------------**
This MR cleans up EmitC source and test files.

**Changes**:
- `mlir/test/mlir-translate/emitc_classops.mlir` was moved to `mlir/test/Target/Cpp/class.mlir`.
- `mlir/test/Dialect/EmitC/wrap_emitc_func_in_class.mlir` and `mlir/test/Dialect/EmitC/wrap_emitc_func_in_class_noAttr.mlir were moved` to `mlir/test/Dialect/EmitC/transforms.mlir`.
- Test for `emitc.class`, `emitc.field` and `emitc.get_field` was added to `mlir/test/Dialect/EmitC/ops.mlir`.
- Unnecessary `-verify-diagnostics` flags were removed from tests.
- Alphabetical order was restored in `mlir/include/mlir/Dialect/EmitC/IR/EmitC.td` and `mlir/lib/Dialect/EmitC/IR/EmitC.cpp`.
- Case 16 in `bool mlir::emitc::isSupportedFloatType(Type type)` in `EmitC.cpp` was refactored to:
```
case 16:
      if (!llvm::isa<Float16Type, BFloat16Type>(type))
        return false;
      LLVM_FALLTHROUGH;
```
**---------------------------OLD---------------------------**

marbre reviewed (changes requested):
Thanks for taking a shot at cleaning up.

I don't think this a NFC with +1,492 −1,476 lines changed as it isn't easy to review and furthermore makes it harder to find the original commits and their commit message which often provide additional details.

The following is
* > mlir/test/mlir-translate/emitc_classops.mlir was moved to mlir/test/Target/Cpp/class.mlir.
* > Test for emitc.class, emitc.field and emitc.get_field was added to mlir/test/Dialect/EmitC/ops.mlir.
* > Unnecessary -verify-diagnostics flags were removed from tests.

is not controversial from my point of view and can be included in a single commit.

To follow https://llvm.org/docs/Contributing.html#how-to-submit-a-patch more closely
> * not contain any unrelated changes
> * be an isolated change. Independent changes should be submitted as separate patches as this makes reviewing easier.

the change
* > Case 16 in bool mlir::emitc::isSupportedFloatType(Type type) in EmitC.cpp was refactored to:

could be submitted on its own even though a one line change.


I don't have a strong opinion on:
* > mlir/test/Dialect/EmitC/wrap_emitc_func_in_class.mlir and mlir/test/Dialect/EmitC/wrap_emitc_func_in_class_noAttr.mlir were moved to mlir/test/Dialect/EmitC/transforms.mlir.

but I think
> * Alphabetical order was restored in mlir/include/mlir/Dialect/EmitC/IR/EmitC.td and mlir/lib/Dialect/EmitC/IR/EmitC.cpp.

should be dropped as there is not much benefit compared to the disadvantage.

EtoAndruwa commented:
Thank you for the review!

> I don't have a strong opinion on: 
> "mlir/test/Dialect/EmitC/wrap_emitc_func_in_class.mlir and mlir/test/Dialect/EmitC/wrap_emitc_func_in_class_noAttr.mlir were moved to mlir/test/Dialect/EmitC/transforms.mlir."

- Perhaps we should rename `transfrom.mlir` to `form-expression.mlir`, since it is not related to the `transform` dialect and does not use the `--transform-interpreter` option. This naming better reflects the purpose of the test, in my opinion.

- Both `wrap_emitc_func_in_class.mlir` and `wrap_emitc_func_in_class_noAttr.mlir` can be combined into `wrap-in-class.mlir`. This would avoid creating extra files while still achieving the same result.

EtoAndruwa commented:
> the change
Case 16 in bool mlir::emitc::isSupportedFloatType(Type type) in EmitC.cpp was refactored to:

OK, I will prepare separate patch for this change.

EtoAndruwa commented:
> > the change
> > Case 16 in bool mlir::emitc::isSupportedFloatType(Type type) in EmitC.cpp was refactored to:
> 
> OK, I will prepare separate patch for this change.

The appropriate patch https://github.com/llvm/llvm-project/pull/152464.

EtoAndruwa committed:
[mlir][emitc] Clean up EmitC tests

Changes:
    - Unnecessary `-verify-diagnostics` flags were removed from tests.
    - `mlir/test/mlir-translate/emitc_classops.mlir` was moved to
      `mlir/test/Target/Cpp/class.mlir`.
    - The `transfrom.mlir` test was renamed to `form-expressions.mlir`
      for clarity.
    - Test for `emitc.class`, `emitc.field` and `emitc.get_field` was
      added to `mlir/test/Dialect/EmitC/ops.mlir`.
    - `wrap_emitc_func_in_class.mlir` and `wrap_emitc_func_in_class_noAttr.mlir`
      were combined into `wrap-func-in-class.mlir`.

jacquesguan reviewed (approved):
LGTM

marbre reviewed (approved):
Thanks for the clean up!
